frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -62,6 +62,30 @@
         vm.subErrors = {};
 
         utilities.showLoader();
+        
+        // get unique rank number from the url & if exists hightlight the entry
+        var elementId = $location.absUrl().split('?')[0].split('#')[1];
+        if (elementId) {
+            $timeout(function () {
+                $anchorScroll(elementId);
+                $scope.currentEntryLink = elementId;
+                $anchorScroll.yOffset = 90;
+                $scope.isHighlight = elementId.split("leaderboardrank-")[1];
+            }, 2000);
```

Comment:
Why is this too in a timeout??

Reply:
Remove this from timeout and trigger this function once the leaderboard is loaded i.e., in on-success

Reply:
@galipremsagar hey, as it's still not working after moving to the `get leaderboard` function so I moved the code to the polling section of leaderboard as it is sure that polling will start after the leaderboard is loaded and then called my code by putting inside a function only once when polling starts, please check if it's valid and working on your side